tests/unit_tests/device/cuda/container_cuda.cpp

```diff
@@ -9,64 +9,303 @@
 #include "container_cuda_kernel.hpp"
 
 // Vecmem include(s)
+#include <vecmem/memory/cuda/device_memory_resource.hpp>
 #include <vecmem/memory/cuda/managed_memory_resource.hpp>
+#include <vecmem/memory/host_memory_resource.hpp>
+#include <vecmem/utils/cuda/copy.hpp>
 
 // GTest include(s)
 #include <gtest/gtest.h>
 
 // System include(s)
+#include <algorithm>
+#include <iterator>
 #include <numeric>
 
 using namespace detray;
 
-TEST(container_cuda, multi_type_store) {
+constexpr double tol{1e-6};
 
-    // Vecmem memory resource
-    vecmem::cuda::managed_memory_resource resource;
+/// Test the access to the single store by summing the contained values
+TEST(container_cuda, single_store) {
+
+    // Vecmem memory resources
+    vecmem::host_memory_resource host_mr;
+    vecmem::cuda::managed_memory_resource mng_mr;
+    vecmem::cuda::device_memory_resource dev_mr;
+    vecmem::cuda::copy cpy;
+
+    // Create single store(s)
+    single_store_t store(host_mr);
+    single_store_t mng_store(mng_mr);
+
+    // Base store function check
+    EXPECT_TRUE(store.empty());
+    EXPECT_EQ(store.size(), 0u);
+    EXPECT_TRUE(mng_store.empty());
+    EXPECT_EQ(mng_store.size(), 0u);
+
+    // Test the managed memory allocation
+    empty_context ctx{};
+    mng_store.reserve(4, ctx);
+    mng_store.emplace_back(ctx, 1.);
+    mng_store.push_back(2., ctx);
+    mng_store.insert(vecmem::vector<double>{10.5, 7.6}, ctx);
+    store.append(mng_store, ctx);
+
+    EXPECT_FALSE(store.empty());
+    EXPECT_EQ(store.size(), 4u);
+    EXPECT_FALSE(mng_store.empty());
+    EXPECT_EQ(mng_store.size(), 4u);
+
+    // Check the host-side access to the data
+    EXPECT_NEAR(mng_store[0], 1., tol);
+    EXPECT_NEAR(mng_store[2], 10.5, tol);
+    EXPECT_NEAR(mng_store.at(1, ctx), 2., tol);
+    EXPECT_NEAR(mng_store.at(3, ctx), 7.6, tol);
+
+    // CPU sum check
+    double cpu_sum{std::accumulate(mng_store.begin(), mng_store.end(), 0.)};
+    EXPECT_NEAR(cpu_sum, 21.1, tol);
+    EXPECT_NEAR(cpu_sum, std::accumulate(store.begin(), store.end(), 0.), tol);
+
+    // Get the view to the managed memory and run the test
+    single_store_t::view_type mng_store_view = detray::get_data(mng_store);
+
+    vecmem::vector<double> cuda_sum(&mng_mr);
+    cuda_sum.push_back(0.);
+    vecmem::data::vector_view<double> sum_data = vecmem::get_data(cuda_sum);
+
+    test_single_store(mng_store_view, sum_data);
+
+    EXPECT_NEAR(cpu_sum, cuda_sum[0], tol);
+
+    // Check that the store can be cleared
+    mng_store.clear(ctx);
+
+    EXPECT_TRUE(mng_store.empty());
+    EXPECT_EQ(mng_store.size(), 0u);
+
+    // Reset for next test
+    cuda_sum[0] = 0.;
+
+    // Copy the host store to device, get the view to it and run the test again
+    single_store_t::buffer_type store_buffer =
+        detray::get_buffer(store, dev_mr, cpy);
+    single_store_t::view_type buffer_view = detray::get_data(store_buffer);
+
+    test_single_store(buffer_view, sum_data);
```

Comment:
You're missing checking that these results are valid.
```suggestion
    test_single_store(buffer_view, sum_data);
    single_store_t::buffer_type host_cp = detray::get_buffer(buffer_view, host_mr, cpy);
    EXPECT_NEAR(cpu_sum, host_cp[0], tol);
```

Reply:
The same applies to all the other tests

Reply:
The result is stored in the ```cuda_sum``` vector, which right now uses managed memory, not in the container. We don't need (even should prevent) write access to the detector on device. I thought I could just (ab)use managed memory like I do?

Reply:
Oh right, sorry. I didn't read into this correctly. It's correct